### PR TITLE
Add a statusMessage to the PolicySet status

### PR DESCRIPTION
**Pull request by gparvin**
The PolicySet status sometimes shows the compliance status and
sometimes it doesn't.  To help understand what's going on we need a
message.  In the future we may have a pending state for policies, but
for now we just want a message in the PolicySet to explain whether the
status is unable to account for some of the policies.

Refs:
 - https://github.com/stolostron/backlog/issues/21750

Signed-off-by: Gus Parvin <gparvin@redhat.com>

### CODE REVIEW

File: controllers/policyset/policyset_controller.go [hunk 1]
```diff
@@ -225,19 +229,48 @@ func (r *PolicySetReconciler) processPolicySet(ctx context.Context, plcSet *poli
...
-			return false
-		} else if policy == "found" {
-			show = true
+	if len(disabledPolicies) > 0 {
+		statusMessage = fmt.Sprintf("Disabled policies: %v", disabledPolicies)
+		separator = "; "
+	}
+
+	if len(pendingPolicies) > 0 {
+		if len(separator) == 0 {
```
Comment (mprahl) [inline controllers/policyset/policyset_controller.go:243]:
```suggestion
		if separator == "" {
```

File: controllers/policyset/policyset_controller.go [hunk 2]
```diff
@@ -225,19 +229,48 @@ func (r *PolicySetReconciler) processPolicySet(ctx context.Context, plcSet *poli
...
+	statusMessage := "All policies are reporting status"
+	separator := ""
 
-	for _, policy := range compliancesFound {
-		if policy == "notfound" {
-			return false
-		} else if policy == "found" {
-			show = true
+	if len(disabledPolicies) > 0 {
+		statusMessage = fmt.Sprintf("Disabled policies: %v", disabledPolicies)
```
Comment (mprahl) [inline controllers/policyset/policyset_controller.go:238]:
```suggestion
		statusMessage = fmt.Sprintf("Disabled policies: %s", strings.Join(disabledPolicies, ", "))
```

File: controllers/policyset/policyset_controller.go [hunk 3]
```diff
@@ -225,19 +229,48 @@ func (r *PolicySetReconciler) processPolicySet(ctx context.Context, plcSet *poli
...
+		separator = "; "
+	}
+
+	if len(pendingPolicies) > 0 {
+		if len(separator) == 0 {
+			statusMessage = ""
+		}
+
+		statusMessage = fmt.Sprintf(statusMessage+separator+
+			"No status provided while policies are pending: %v", pendingPolicies)
```
Comment (mprahl) [inline controllers/policyset/policyset_controller.go:248]:
```suggestion
			"No status provided while policies are pending: %s", strings.Join(pendingPolicies, ", "))
```

File: controllers/policyset/policyset_controller.go [hunk 4]
```diff
@@ -225,19 +229,48 @@ func (r *PolicySetReconciler) processPolicySet(ctx context.Context, plcSet *poli
...
+		separator = "; "
+	}
+
+	if len(deletedPolicies) > 0 {
+		if len(separator) == 0 {
```
Comment (mprahl) [inline controllers/policyset/policyset_controller.go:253]:
```suggestion
		if separator == "" {
```

File: controllers/policyset/policyset_controller.go [hunk 5]
```diff
@@ -225,19 +229,48 @@ func (r *PolicySetReconciler) processPolicySet(ctx context.Context, plcSet *poli
...
+			statusMessage = ""
 		}
+
+		statusMessage = fmt.Sprintf(statusMessage+separator+"Deleted policies: %v", deletedPolicies)
```
Comment (mprahl) [inline controllers/policyset/policyset_controller.go:257]:
```suggestion
		statusMessage = fmt.Sprintf(statusMessage+separator+"Deleted policies: %s", strings.Join(deletedPolicies, ", "))
```

**Comment by mprahl** [comment]
@gparvin could you please add a test case that has pending, disabled, and compliant policies in the same policy set? That would test the status message concatenation logic.

**Comment by gparvin** [comment]
@mprahl I've addressed the review comments including adding a new test that includes a pending policy.  Thanks for that idea.

**Review by mprahl — APPROVED** [review]
This looks great!

/hold in case you want other reviewers.

**Pull request merged**